tests/components/poolsense/test_config_flow.py

```diff
@@ -0,0 +1,55 @@
+"""Test the PoolSense config flow."""
+from homeassistant import data_entry_flow
+from homeassistant.components.poolsense.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+
+from tests.async_mock import patch
+
+
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == SOURCE_USER
+
+
+async def test_invalid_credentials(hass):
+    """Test we handle invalid credentials."""
+    with patch(
+        "poolsense.PoolSense.test_poolsense_credentials", return_value=False,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+            data={CONF_EMAIL: "test-email", CONF_PASSWORD: "test-password"},
+        )
+
+    assert result["type"] == "form"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5760157)